tests/test_templates.py

```diff
@@ -37,282 +47,308 @@
-        def circuit(**inp_):
-            qml.Displacement(1., 1., wires=0)
-            template(*inp_.values(), wires=range(2))
-            template(*inp_.values(), wires=range(2))
-            qml.Displacement(1., 1., wires=1)
-            return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
-
-        circuit(**inpts)
-
-
-class TestIntegrationCircuitTf:
-    """Tests the integration of templates into circuits using the TensorFlow interface."""
-
-    @pytest.mark.parametrize(fixture, qubit_const)
-    def test_integration_qubit_positional_arg(self, template, inpts):
-        """Checks integration of qubit templates using positional arguments to qnode."""
-
-        inpts = [tf.Variable(i) for i in inpts]
-        dev = qml.device('default.qubit', wires=2)
-
-        @qml.qnode(dev, interface='tf')
-        def circuit(*inp_):
-            qml.PauliX(wires=0)
-            template(*inp_, wires=range(2))
-            template(*inp_, wires=range(2))
-            qml.PauliX(wires=1)
-            return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
-
-        circuit(*inpts)
-
-    @pytest.mark.parametrize(fixture, qubit_const)
-    def test_integration_qubit_keyword_arg(self, template, inpts):
-        """Checks integration of qubit templates using keyword arguments to qnode."""
-
-        inpts = {"w"+str(1): tf.Variable(inpts[i]) for i in range(len(inpts))}
-        dev = qml.device('default.qubit', wires=2)
-
-        @qml.qnode(dev, interface='tf')
-        def circuit(**inp_):
-            qml.PauliX(wires=0)
-            template(*inp_.values(), wires=range(2))
-            template(*inp_.values(), wires=range(2))
-            qml.PauliX(wires=1)
-            return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
-
-        circuit(**inpts)
-
-    @pytest.mark.parametrize(fixture, cv_const)
-    def test_integration_cv_positional_args(self, gaussian_device_2_wires, template, inpts):
-        """Checks integration of continuous-variable templates using positional arguments to qnode."""
-
-        inpts = [tf.Variable(i) for i in inpts]
+            return qml.expval(qml.Identity(0))
 
-        @qml.qnode(gaussian_device_2_wires, interface='tf')
+        # Check gradients in numpy interface
+        if intrfc == 'numpy':
+            grd = qml.grad(circuit, argnum=argnm)
+            res_numpy = grd(*inpts)
+
+        # Check gradients in torch interface
+        if intrfc == 'torch':
+            for a in argnm:
+                inpts[a] = TorchVariable(inpts[a], requires_grad=True)
+            res = circuit(*inpts)
+            res.backward()
+            res_torch = [inpts[a].grad.numpy() for a in argnm]
+
+        # Check gradients in tf interface
+        if intrfc == 'tf':
+            grad_inpts = [inpts[a] for a in argnm]
+            with tf.GradientTape() as tape:
+                loss = circuit(*inpts)
+                res_tf = tape.gradient(loss, grad_inpts)
```

Comment:
What is being asserted in this test? Just that the gradient functions execute without error?

Reply:
Yes, but I adapted the design to avoid the redundant new variable name